files/galaxy-test/dynamic_rules/usegalaxy/sorting_hat.py

```diff
@@ -147,24 +171,41 @@ def name_it(tool_spec):
     if 'name' in tool_spec:
         name += '_' + tool_spec['name']
 
+    # Force a replacement of the destination's id
+    if tool_spec.get('force_destination_id', False):
+        name = prefix + tool_spec.get('runner')
+
     return name
 
 
-def _get_limits(destination, default_cores=1, default_mem=4, default_gpus=0):
+def _get_limits(destination, dest_spec=SPECIFICATIONS, default_cores=1, default_mem=4, default_gpus=0):
     limits = {'cores': default_cores, 'mem': default_mem, 'gpus': default_gpus}
-    limits.update(SPECIFICATIONS.get(destination).get('limits', {}))
+    limits.update(dest_spec.get(destination).get('limits', {}))
     return limits
 
 
-def build_spec(tool_spec, runner_hint=None):
-    destination = tool_spec.get('runner', 'condor')
+def _weighted_random_sampling(destinations, dest_spec=SPECIFICATIONS):
+    bunch = []
+    for d in destinations:
+        weight = SPECIFICATIONS[d].get('nodes', 1)
```

Comment:
should this be `dest_spec`

Reply:
yes, it should. I will fix it after the update